### PR TITLE
Add redis db index option

### PR DESCRIPTION
**Pull request by inuyasha82**
This PR  let the user choose what database to use  instead of hardcoding db index 0. 

What is needed is to  specify: 

`dbIdx: X`

value in the metadata,

### CODE REVIEW

File: pkg/scalers/redis_scaler.go [hunk 1]
```diff
@@ -30,6 +31,7 @@ type redisMetadata struct {
 	listName         string
 	address          string
 	password         string
+	dbIdx            int
```
Comment (tomkerkhove) [inline pkg/scalers/redis_scaler.go]: Let's rename this to `databaseIndex` maybe?
```suggestion
	databaseIndex            int
```

Will be easier for new folks to understand what it represents imo

**Comment by marchmallow** [comment]
Guess we should document also the new option for TLS connections in https://github.com/kedacore/keda-docs/blob/master/content/scalers/redis-lists.md

**Comment by inuyasha82** [comment]
@tomkerkhove @marchmallow yeah, i'll update the documentation today, and also make the required changes for this PR.

**Comment by inuyasha82** [comment]
Ok i merged with master, and updated the database name. 

I tested on databaseIndex 1 and it is working. 
I also added the default value for enableTLS.

**Comment by inuyasha82** [comment]
Create PR for the documentation too: https://github.com/kedacore/keda-docs/pull/74

**Pull request merged**